llvm/lib/SYCLLowerIR/SYCLPropagateAspectsUsage.cpp

```diff
@@ -369,12 +403,28 @@ buildFunctionsToAspectsMap(Module &M, TypeToAspectsMapTy &TypesWithAspects) {
   }
 
   SmallPtrSet<const Function *, 16> Visited;
-  for (Function *F : EntryPoints)
+  for (Function *F : EntryPoints) {
+    setSyclFixedTargetsMD(F, Targets);
```

Comment:
I would prefer this to be done directly at a higher level, because this functionality is not a part of building functions to aspects map. 

To avoid gathering entry points in two places, we can move the code collecting entry points to a higher level as well and then accept `EntryPoints` here as an argument.

Reply:
Done [8929df0](https://github.com/intel/llvm/pull/7115/commits/8929df061795a348c71cf220a48024129a5732fd)